File: Java/ModularService/ModularLibrary/src/main/java/com/hydron/sample/modularlibrary/pojo/Bar.java
package com.hydron.sample.modularlibrary.pojo;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.*;

import java.time.Instant;

@NoArgsConstructor
@AllArgsConstructor
@Builder
@Getter
@ToString
@JsonInclude(JsonInclude.Include.NON_NULL)
public class Bar implements ExampleData {
    @JsonProperty
    private String name;

    @JsonProperty
    private String identifier;

    @JsonProperty
    private Instant timestamp;

    @JsonProperty
    private static final String text = "%s [%s] - Bar says \"Goodbye, %s!\"";

    @Override
    public String getMessage() {
        return String.format(text, this.identifier, this.timestamp, this.name);
    }
}

File: Rust/modular_library/src/bar.rs
use crate::example_data::ExampleData;

pub struct Bar {
    pub name: String,
    pub identifier: String,
    pub timestamp: String,
}

impl ExampleData for Bar {
    fn new(name: &str, identifier: &str, timestamp: String) -> Self {
        return Self {
            name: String::from(name),
            identifier: String::from(identifier),
            timestamp,
        };
    }

    fn get_message(&self) -> String {
        return format!("{} [{}] - Bar says \"Goodbye, {}!\"", self.identifier, self.timestamp, self.name);
    }
}
